vortex-scalar/src/lib.rs

```diff
@@ -100,39 +100,38 @@ impl Scalar {
         }
     }
 
-    pub fn cast(&self, dtype: &DType) -> VortexResult<Self> {
-        if self.is_null() && !dtype.is_nullable() {
-            vortex_bail!("Can't cast null scalar to non-nullable type")
+    pub fn cast(&self, target: &DType) -> VortexResult<Self> {
+        if let DType::Extension(ext_dtype) = target {
+            let storage_scalar = self.cast_to_non_extension(ext_dtype.storage_dtype())?;
```

Comment:
I think it's theoretically possible right now to have a storage dtype that is an Extension type... we should potentially prohibit that

Reply:
Indeed we [prohibit that](https://github.com/spiraldb/vortex/blob/develop/vortex-dtype/src/extension.rs#L101-L103).